src/get-started/test-service.md

```diff
@@ -0,0 +1,352 @@
+---
+title: Service testing tutorial
+---
+# Service Testing Tutorial
+
+As blockchain technology is focused on security, a natural desire after
+[creating an Exonum service](create-service.md) is to test it. This tutorial
+shows how to accomplish this task with the help of [the testkit library](../advanced/service-testing.md).
+
+## Preparing for Integration Testing
+
+Recall that an Exonum service is typically packaged as a Rust crate. Correspondingly,
+service testing could naturally be performed with the help of [integration tests][integration-testing],
+in which the service is treated like a black box. In the case of the cryptocurrency
+service, which we created in the previous tutorial, it would be natural to test
+how the service reacts to overcharges, transfers from or to the unknown wallet,
+transfers to self, and other scenarios which may work not as expected.
+
+Exonum has a handy tool in its disposal to test services in this manner –
+the [**exonum-testkit**](../advanced/service-testing.md) crate. To use it,
+we need to add the following lines to the project’s `Cargo.toml`:
+
+```toml
+[dev-dependencies]
+exonum-testkit = "0.1.0"
+```
+
+Then, let’s create [`tests/api.rs`][tests-api.rs], a file which will
+contain the tests.
+
+## Imports
+
+The created file is executed separately from the service code, meaning that we
+need to import the service crate along with **exonum** and **exonum-testkit**:
+
+```rust
+extern crate cryptocurrency;
+extern crate exonum;
+extern crate exonum_testkit;
+```
+
+Just like with the service itself, we then import types we will use:
+
+```rust
+use exonum::crypto::{self, PublicKey, SecretKey};
+use exonum::messages::Message;
+use exonum_testkit::{ApiKind, TestKit, TestKitApi, TestKitBuilder};
+
+// Import datatypes used in tests from the crate where the service is defined.
+use cryptocurrency::{TxCreateWallet, TxTransfer, TransactionResponse,
+                     Wallet, CurrencyService};
+```
+
+## API Wrapper
```

Comment:
And will not to be simpler to create a `CryptocurrencyApi` trait and implement it for the `TestKitApi`?